examples/esp32/esp32-server.rs

```diff
@@ -93,25 +90,56 @@ mod esp32 {
             (wifi.wifi().sta_netif().get_ip_info().unwrap().ip, wifi)
         };
 
-        let cfg = AppClientConfig::new(
-            ROBOT_SECRET.to_owned(),
-            ROBOT_ID.to_owned(),
-            ip,
-            "".to_owned(),
-        );
-        let webrtc_certificate = WebRtcCertificate::new(
-            ROBOT_DTLS_CERT.to_vec(),
-            ROBOT_DTLS_KEY_PAIR.to_vec(),
-            ROBOT_DTLS_CERT_FP,
-        );
-
-        let tls_cfg = {
-            let cert = [ROBOT_SRV_PEM_CHAIN.to_vec(), ROBOT_SRV_PEM_CA.to_vec()];
-            let key = ROBOT_SRV_DER_KEY;
-            Esp32TLSServerConfig::new(cert, key.as_ptr(), key.len() as u32)
-        };
-
-        serve_web(cfg, tls_cfg, repr, ip, webrtc_certificate, max_connection);
+        #[cfg(not(feature = "provisioning"))]
+        {
+            use micro_rdk::{
+                common::app_client::AppClientConfig,
+                esp32::{certificate::WebRtcCertificate, tls::Esp32TLSServerConfig},
+            };
+            let cfg = AppClientConfig::new(
+                ROBOT_SECRET.to_owned(),
+                ROBOT_ID.to_owned(),
+                ip,
+                "".to_owned(),
+            );
+            let webrtc_certificate = WebRtcCertificate::new(
+                ROBOT_DTLS_CERT.to_vec(),
+                ROBOT_DTLS_KEY_PAIR.to_vec(),
+                ROBOT_DTLS_CERT_FP,
+            );
+
+            let tls_cfg = {
+                let cert = [ROBOT_SRV_PEM_CHAIN.to_vec(), ROBOT_SRV_PEM_CA.to_vec()];
+                let key = ROBOT_SRV_DER_KEY;
+                Esp32TLSServerConfig::new(cert, key.as_ptr(), key.len() as u32)
+            };
+            micro_rdk::esp32::entry::serve_web(
+                cfg,
+                tls_cfg,
+                repr,
+                ip,
+                webrtc_certificate,
+                max_connection,
+            );
+        }
+        #[cfg(feature = "provisioning")]
+        {
+            use micro_rdk::common::provisioning::{
+                server::ProvisioningInfo, storage::MemoryCredentialStorage,
+            };
+            let mut info = ProvisioningInfo::default();
+            info.set_fragment_id("d385b480-3d19-4fad-a928-b5c18a58d0ed".to_string());
+            info.set_manufacturer("viam".to_owned());
+            info.set_model("test-esp32".to_owned());
```

Comment:
Yes we should but I would like to keep it to a separate ticket, we don't know for now how we are going to let user do provisioning (we wait for the mobile app for that) Since its gated behind a feature we don't have to integrate perfectly. The fragment is public and contains just a fake board